### PR TITLE
[SPARK-43774][BUILD] Upgrade FasterXML jackson to 2.15.1

### PR DESCRIPTION
**Pull request by ronandoolan2**
### What changes were proposed in this pull request?
Upgrade FasterXML jackson from 2.15.0 to 2.15.1

### Why are the changes needed?
New version that fix some bugs, the full release-notes as follows:

- https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.15.1 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GA

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,3 +123,4 @@ in the online documentation for an overview on how to configure Spark.
 
 Please review the [Contribution to Spark guide](https://spark.apache.org/contributing.html)
 for information on how to get started contributing to the project.
+##
```
Comment (LuciferYang) [inline README.md:126]: hmm... why we need this change?

Comment (ronandoolan2) [reply README.md:126]: I had trouble retriggering the job after enabling github actions on my fork, I'll remove this line again

**Comment by ronandoolan2** [comment]
Ok my scans were giving a spurious result sorry

**Comment by bjornjorgensen** [comment]
@ronandoolan2 you can update it to 2.15.1 if you will?

**Comment by ronandoolan2** [comment]
ok sure

**Comment by bjornjorgensen** [comment]
we need a JiRA for this 
and you must run `./dev/test-dependencies.sh --replace-manifest` to update the rest of pom.xml file.

**Comment by bjornjorgensen** [comment]
This is how I wood have done it.. 
Feel free to change what you will change :) 

Change tittle to something like [SPARK-XXXX][BUILD] Upgrade FasterXML jackson to 2.15.1


What changes were proposed in this pull request?
Upgrade FasterXML jackson from 2.15.0 to 2.15.1

Why are the changes needed?
New version that fix some bugs see release-notes https://github.com/FasterXML/jackson-databind/blob/77789abaecd2e42a3765af5231e252ee62578b18/release-notes/VERSION-2.x#LL17C21-L17C21 for more info.

Does this PR introduce any user-facing change?
No.

How was this patch tested?
Pass GA

**Comment by LuciferYang** [comment]
Let update the release notes to https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.15.1

And if upgrading to this version has benefits for Spark, it can also be listed in the PR description. @ronandoolan2

**Comment by bjornjorgensen** [comment]
@ronandoolan2 you need to have a JIRA account https://selfserve.apache.org/jira-account.html
for this one you can use https://issues.apache.org/jira/browse/SPARK-43774 witch is what you are using.

**Comment by ronandoolan2** [comment]
> Let's update the release notes to https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.15.1
> 
> And if upgrading to this version has benefits for Spark, it can also be listed in the PR description. @ronandoolan2

Where are the release notes that I need to update, I couldn't find them in the repo

**Comment by bjornjorgensen** [comment]
@ronandoolan2 It looks like @LuciferYang  have changed it. 

"New version that fix some bugs, the full release-notes as follows:

https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.15.1"

**Comment by ronandoolan2** [comment]
> @ronandoolan2 It looks like @LuciferYang have changed it.
> 
> "New version that fix some bugs, the full release-notes as follows:
> 
> https://github.com/FasterXML/jackson/wiki/Jackson-Release-2.15.1"

Ok great thanks, do you know when the next version of pyspark will be released to pypi?

**Review by LuciferYang — APPROVED** [review]

**Review by srowen — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged to master. Thanks @ronandoolan2 @srowen @bjornjorgensen  ~

**Comment by bjornjorgensen** [comment]
Welcome, @ronandoolan2 

We are thrilled to have you join the Apache Spark community as a new contributor. It's always exciting to see fresh minds and ideas. We're confident that your skills and expertise will be an asset to our project, enhancing our collaborative efforts to improve Apache Spark.

Just as a reminder, the Apache Spark community values open communication, respect, and inclusiveness. Don't hesitate to ask questions, share your thoughts, or request help when needed. Whether it's through bug fixes, adding new features, or improving documentation, every contribution is highly appreciated.

We hope that you'll find this experience rewarding. Together, we can make Apache Spark even better. Again, welcome aboard, @ronandoolan2

**Pull request closed without merging**